sdk/main/src/AccountUpdateTransaction.cc

```diff
@@ -128,6 +136,15 @@ AccountUpdateTransaction& AccountUpdateTransaction::setStakedAccountId(const Acc
   return *this;
 }
 
+//-----
+AccountUpdateTransaction& AccountUpdateTransaction::clearStakedAccountId()
+{
+  requireNotFrozen();
+  mStakedAccountId = AccountId(0ULL);
```

Comment:
Since `mStakedAccountId` is optional, shouldn't we set it to `std::nullopt` when clearing the `StakedAccountId`?

Reply:
Same explanation applies here as it does for [`mAccountMemo`](https://github.com/hashgraph/hedera-sdk-cpp/pull/558#discussion_r1368755086).

Reply:
Thanks for the explanation. I agree this implementation is the correct one.